packages/config/src/discovery/ProjectDiscovery.ts

```diff
@@ -744,24 +744,29 @@ export class ProjectDiscovery {
           return (
             value.permission +
             ':' +
-            (value.via !== undefined ? formatVia(value.via) : '')
+            (value.via !== undefined ? formatVia(value.via) : '') +
+            ':' +
+            (value.description ?? '')
           )
         },
       ),
     ).map(([key, entries]) => {
       const permission = key.split(':')[0] as PermissionType
       const via = key.split(':')[1] ?? ''
+      const description = key.split(':', 3)[2] ?? ''
```

Comment:
Wouldn't it be better to return an object above and just do `.description`?

Reply:
Can't do it, because the concatenated string is used in `groupBy` - it's the key and it can't be an object. This makes it really simple to group projects that share the same description, path and delay. It's not the prettiest code, but gets thing done in a few simple lines of code.